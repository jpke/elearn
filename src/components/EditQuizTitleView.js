import React from 'react';

const EditQuizTitleView = (props) =>  {
  console.log("edit quiz view :" , props);
return (
    <div className="editQuizTitle">
      <input type="text" id="0 title" name="title" className="edit-quiz-item" defaultValue={props.title} onChange={(e) => {props.editQuizItem(e)}}/>
      <input type="number" id="0 minimumScore" name="minimumScore" className="edit-quiz-item" defaultValue={props.minimumScore} onChange={(e) => {props.editQuizItem(e)}}/>
    </div>
);
}

export default EditQuizTitleView;
