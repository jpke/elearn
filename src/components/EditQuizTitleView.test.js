/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';

import EditQuizTitleView from './EditQuizTitleView';

it('renders correctly', () => {
  const tree = renderer.create(
    <EditQuizTitleView
      title={"Sample Quiz Title"}
      minimumScore={1}
      editQuizItem= {jest.fn()}
      saveQuiz={jest.fn()}
      deleteQuiz={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
