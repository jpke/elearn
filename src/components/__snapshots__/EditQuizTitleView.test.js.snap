exports[`test renders correctly 1`] = `
<div
  className="editQuizTitle">
  <h3>
    Title
  </h3>
  <input
    className="edit-quiz-item"
    id="0 title"
    name="title"
    onChange={[Function]}
    type="text"
    value="Sample Quiz Title" />
  <section>
    <div
      className="delete">
      <button
        className="deleteQuiz"
        id="deleteQuiz"
        onClick={[Function]}>
        Delete
      </button>
    </div>
    <div>
      <div
        className="passing-score">
        <h3>
          Passing Score
        </h3>
        <input
          className="edit-minimumScore"
          id="0 minimumScore"
          name="minimumScore"
          onChange={[Function]}
          type="number"
          value={1} />
      </div>
    </div>
    <div
      className="save">
      <button
        className="updateQuiz"
        id="updateQuiz"
        onClick={[Function]}>
        Save
      </button>
    </div>
  </section>
</div>
`;
